libs/log/sem_logger.go

```diff
@@ -0,0 +1,176 @@
+package log
+
+// Selective Entity Monitor (SEM) Logger enables rule based log traces
+
+type SemLogger interface {
+	Logger
+	Entry(SemEntity, []byte)
+	Exit(SemEntity, []byte)
+	AddRule(SemEntity, []byte)
+	DeleteAll()
+	Status() (SemStatus, error)
+}
+
+// type loglevel byte
+type SemEntity int
+
+const (
+	SemTransaction SemEntity = iota + 1
+	SemAddress
+)
+
+type SemStatus struct {
+	Active bool
+	Rules  map[SemEntity]map[string]bool
```

Comment:
TODO: make this shared object across multiple SEM logger instances to be able to cope with on the fly updates on rules

Reply:
Should it be the one in line 29?

Reply:
yes